examples/AdvancedOperations/AddGmailAd.php

```diff
@@ -0,0 +1,251 @@
+            /** @var MediaFile $addedMediaFile */
+            printf(
+                "Created media file with resource name '%s'.%s",
+                $addedMediaFile->getResourceName(),
+                PHP_EOL
+            );
+        }
+
+        // Returns the created media file resource names.
+        return [
+            'logoResourceName' => $response->getResults()[0]->getResourceName(),
+            'marketingImageResourceName' => $response->getResults()[1]->getResourceName()
+        ];
+    }
+
+    /**
+     * Adds the gmail ad.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the client customer ID
+     * @param int $adGroupId the ad group ID
+     * @param array $mediaFiles the media file resource names
+     */
+    private static function addGmailAd(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        int $adGroupId,
+        array $mediaFiles
+    ) {
+        // Creates the Gmail ad info.
+        $gmailAdInfo = new GmailAdInfo([
+            // Sets the teaser information.
+            'teaser' => new GmailTeaser([
+                'headline' => new StringValue(['value' => 'Dream']),
+                'description' => new StringValue(['value' => 'Create your own adventure']),
+                'business_name' => new StringValue(['value' => 'Interplanetary Ships']),
+                'logo_image' => new StringValue(['value' => $mediaFiles['logoResourceName']])
+            ]),
+            // Sets the marketing image and other information.
+            'marketing_image' =>
+                new StringValue(['value' => $mediaFiles['marketingImageResourceName']]),
+            'marketing_image_headline' => new StringValue(['value' => 'Travel']),
+            'marketing_image_description' => new StringValue(['value' => 'Take to the skies!'])
+        ]);
+
+        // Sets the Gmail ad info on an Ad.
+        $ad = new Ad([
+            'name' => new StringValue(['value' => "Gmail Ad #" . uniqid()]),
+            'final_urls' => [new StringValue(['value' => 'http://www.example.com'])],
+            'gmail_ad' => $gmailAdInfo
+        ]);
+
+        // Gets the ad group resource name.
+        $adGroupResourceName =
+            new StringValue(['value' => ResourceNames::forAdGroup($customerId, $adGroupId)]);
+
+        // Creates an ad group ad to hold the above ad.
+        $adGroupAd = new AdGroupAd([
+            'ad' => $ad,
+            'status' => AdGroupAdStatus::PAUSED,
+            'ad_group' => $adGroupResourceName
+        ]);
+
+        // Creates an ad group ad operation.
+        $adGroupAdOperation = new AdGroupAdOperation();
+        $adGroupAdOperation->setCreate($adGroupAd);
+
+        // Issues a mutate request to add the ad group ad.
+        $adGroupAdServiceClient = $googleAdsClient->getAdGroupAdServiceClient();
+        $response = $adGroupAdServiceClient->mutateAdGroupAds($customerId, [$adGroupAdOperation]);
```

Comment:
Should we explicitly handle the cases where the provided campaignId and/or adGroupId don't allow the creation of a GmailAd?

Reply:
The prerequisite is explained in the description of the class and I could not find any other code example doing such handling so I do not think it is required here.

But I agree that it could be interesting to demonstrate how to handle errors like this one, may be a separated code example in the "ErrorHandling" session would be a better fit?

Reply:
The add dynamic feed example does something similar [here](https://github.com/googleads/google-ads-php/pull/82/files#diff-7a17d89c2c186ebdd6a49a2eb5280749R411)

Reply:
Would a test run at the beginning of the example that explicitly makes a request to retrieve the ad group information and throw an exception if invalid be good?

Reply:
Mmm not sure about that as it would probably add some code that is not really relevant to the purpose of the example. I'd rather just let the thrown exception bubble up and break the example execution in case the campaign or ad group don't support GmailAd.

Reply:
I tried with a bad campaign type and an exception already bubbles up with a "OPERATION_NOT_PERMITTED_FOR_CAMPAIGN_TYPE" error code and some logs:
-----
Request with ID 'XXX' has failed.
Google Ads failure details:
	operation_access_denied_error: This operation is not permitted on this campaign type
-----

This seems pretty adapted so I do not think that catching/parsing the exception and then throwing a new one would add anything to this example?

Reply:
I agree, let's leave it as it is.